### DISCUSSION TITLE
Questions about Git in MSYS2 Proper

### DISCUSSION THREAD

**Discussion by mucookul** [git-for-windows/git#5117] [Q&A] [answered]
First of all, I understand that this is unsupported, I just have a couple questions.

Firstly, what are the known issues when not swapping to gitforwindows's version of msys2-runtime?

Secondly, is it possible to run git credential manager outside of MINGW64? I try to run it but I get: `error: remote-curl: error reading command stream from git`. It works file in MINGW64 though.

**Comment by dscho** [comment, marked as answer]
> Firstly, what are the known issues when not swapping to gitforwindows's version of msys2-runtime?

With https://github.com/msys2/msys2-runtime/pull/182 merged, there are a lot less issues now.

However, there are still a few things that Git for Windows' variant of the MSYS2 runtime does differently, most notably there are these patches, still:

- https://github.com/git-for-windows/msys2-runtime/commit/0ad833be11629ca667a1175e5d6a09be8a5afed7
- https://github.com/git-for-windows/msys2-runtime/commit/2bd5e69217037d93c46c3289eece622f2c571bdf
- https://github.com/git-for-windows/msys2-runtime/commit/10faa7e47c27daf5b988d3425fd6ef7fc68e6a68
- https://github.com/git-for-windows/msys2-runtime/commit/51451dbb2e7951c7b25ed6dac8d137f6bf896b36
- https://github.com/git-for-windows/msys2-runtime/commit/5246c137d6f30077aab548cb7459cffe0325b2f9
- https://github.com/git-for-windows/msys2-runtime/commit/e668c818d69b14443312e7c096bddd92415f8227

All of these change behavior, and we will have to figure out how to upstream them (and where: MSYS2 or Cygwin?).

> is it possible to run git credential manager outside of MINGW64?

Yes.

> I get: `error: remote-curl: error reading command stream from git`.

Which `git.exe` are you using?

**Reply by mucookul** [reply]
The one in `C:\msys64\cmd`

**Reply by mucookul** [reply]
Update: It works when I manually install GCM Core for Windows.
